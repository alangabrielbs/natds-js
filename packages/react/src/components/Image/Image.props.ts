export interface ImageProps {
  /**
   * Specify an optional className to be added to your Image
   */
  className?: string

  /**
   * Optional ID for testing
   */
  testID?: string;

  sourceImage: string

  alternativeText?: string
}
