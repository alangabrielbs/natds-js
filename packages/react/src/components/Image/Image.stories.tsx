import React from 'react'
import { Story, Meta } from '@storybook/react'
import { Image, ImageProps } from '.'

const componentStatus = `
- - -

**NOTE FOR UXs**: This component is available in the following variants:

  - ❌ Standard

With the following attribute statuses:

  - **Attributes:**
    - ❌ \`Types\`

- - -
`

export default {
  title: 'Components/Image',
  component: Image,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
} as Meta

export const Playground: Story<ImageProps> = (args) => (
  <Image {...args} />
)
