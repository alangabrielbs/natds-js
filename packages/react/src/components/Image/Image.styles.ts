import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'

type StyleProps = {
  highlight: boolean,
  fade: 'top' | 'bottom' | 'right' | 'left' | undefined
}

const setBackground = () => ({ highlight, fade }: StyleProps) => {
  if (highlight && !fade) {
    return 'rgba(0, 0, 0, 0.48)'
  }

  if (highlight && fade) {
    return `linear-gradient(to ${fade}, #000, transparent)`
  }

  return 'transparent'
}

const styles = createUseStyles((theme: Theme) => ({
  wrapper: {
    position: 'relative',
    '& > img': {
      width: '100%',
      height: '100%'
    }
  },
  overlay: {
    padding: theme.spacing.micro,
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    background: setBackground()
  }
}))

export default styles
