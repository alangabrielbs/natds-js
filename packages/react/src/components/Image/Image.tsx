import React from 'react'
import styles from './Image.styles'
import { ImageProps } from './Image.props'

const Image = React.forwardRef<HTMLImageElement, ImageProps>(
  (props, ref) => {
    const {
      className = '',
      sourceImage,
      alternativeText = '',
      highlight = false,
      fade,
      children,
      ...rest
    } = props
    const { wrapper, overlay } = styles({ highlight, fade })

    return (
      <div className={`${className} ${wrapper}`}>
        <img
          {...rest}
          ref={ref}
          src={sourceImage}
          alt={alternativeText}
          onError={() => {
            console.log('deu ruim')
          }}
        />
        {highlight && <div className={overlay}>{children}</div>}
      </div>
    )
  }
)

export default Image
