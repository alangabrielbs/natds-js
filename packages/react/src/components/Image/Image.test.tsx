import React from 'react'
import Image from '.'
import renderWithTheme from '../../helpers/renderWithTheme'

describe('Image component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<Image />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
