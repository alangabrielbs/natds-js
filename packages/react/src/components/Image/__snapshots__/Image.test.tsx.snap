// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should render correctly with default props 1`] = `
Array [
  ".wrapper-0-2-1 {
  overflow: hidden;
  position: relative;
}
.wrapper-0-2-1 > img {
  width: 100%;
  height: 100%;
}
.overlay-0-2-2 {
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  padding: 4px;
  position: absolute;
}
.wrapper-d0-0-2-3 {
  border-radius: none;
}
.overlay-d1-0-2-4 {
  background: transparent;
}",
  <div>
    <div
      class=" wrapper-0-2-1 wrapper-d0-0-2-3"
    >
      <img
        alt=""
        src="#"
      />
    </div>
  </div>,
]
`;
